File: backend/app/schemas/summarize.py
"""Schemas for POST /api/v1/summarize (Objective O2 — implemented CP10)."""

from pydantic import BaseModel, Field

from app.schemas.common import ArticleTextRequest


class SummarizeRequest(ArticleTextRequest):
    """Request body for summarization."""


class SummarizeResponse(BaseModel):
    summary: str = Field(..., description="Concise abstractive summary")
    original_length: int = Field(..., ge=0)
    summary_length: int = Field(..., ge=0)

File: frontend/src/types/summarize.ts
/** Mirror of backend/app/schemas/summarize.py (CP10 — Objective O2) */

import type { ArticleTextRequest } from "@/types/common";

export type SummarizeRequest = ArticleTextRequest;

export type SummarizeResponse = {
  summary: string;
  original_length: number;
  cleaned_length: number;
  summary_length: number;
  model: string;
  latency_ms: number;
  prompt_tokens: number | null;
  output_tokens: number | null;
};
